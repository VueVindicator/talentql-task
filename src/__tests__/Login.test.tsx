import React from "react";
import { render, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import Login from "../components/auth/login/login";

describe("<Login />", () => {
  test("should render a login form", async () => {
    const { getByTestId } = render(<Login />);
    const loginForm = await getByTestId("login-form");

    expect(loginForm).toHaveFormValues({
      username: 'theusername',
      password: 'thepassword',
    });
  });

  test("the form should allow a user to enter a username", async () => {
    const { getByTestId } = render(<Login />);

    fireEvent.change(getByTestId("username"), {
      target: { value: "theusername" },
    });

    expect(getByTestId("username")).toHaveValue("theusername");
  });

  test("the form should allow a user to enter a password", async () => {
    const { getByTestId, findByRole } = render(<Login />);

    fireEvent.change(getByTestId("password"), {
      target: { value: "thepassword" },
    });

    expect(getByTestId("password")).toHaveValue("thepassword");
  });

  test("the form should be submitted with username and password", async () => {
    const handleSubmit = jest.fn();
    const { getByTestId } = render(<Login />);
    const { getInstance, toJSON } = renderer.create(<Login />);

    let tree = toJSON();
    expect(tree).toMatchSnapshot();

    fireEvent.change(getByTestId("username"), {
      target: { value: "theusername" },
    });
    fireEvent.change(getByTestId("password"), {
      target: { value: "thepassword" },
    });
    fireEvent.click(getByTestId("submit"));

    expect(getByTestId("submit")).toBeEnabled();
  });
});
