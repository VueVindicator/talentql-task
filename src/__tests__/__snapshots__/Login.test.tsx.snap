// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> the form should be submitted with username and password 1`] = `
<div
  className="sc-bdnxRM cGZbgt"
>
  <div
    className="login-card"
  >
    <form
      data-testid="login-form"
      onSubmit={[Function]}
    >
      <div
        className="sc-gtsrHT gwPHhe"
      >
        <label
          htmlFor="username"
        >
          Username
        </label>
        <input
          data-testid="username"
          name="text"
          onChange={[Function]}
          placeholder="Please enter your Username"
          type="text"
          value="theusername"
        />
      </div>
      <div
        className="sc-gtsrHT gwPHhe"
      >
        <label
          htmlFor="password"
        >
          Password
        </label>
        <input
          data-testid="password"
          name="password"
          onChange={[Function]}
          placeholder="Please enter your Password"
          type="password"
          value="thepassword"
        />
      </div>
      <div
        className="sc-dlnjwi iawPsM"
      >
        <button
          data-testid="submit"
          type="submit"
        >
          Login
        </button>
      </div>
    </form>
  </div>
  <div
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "bottom": 16,
        "left": 16,
        "pointerEvents": "none",
        "position": "fixed",
        "right": 16,
        "top": 16,
        "zIndex": 9999,
      }
    }
  />
</div>
`;
